pkg/kapp/diff/change_set.go

```diff
@@ -23,13 +32,24 @@ type ChangeSet struct {
 func NewChangeSet(existingRs, newRs []ctlres.Resource,
 	opts ChangeSetOpts, changeFactory ChangeFactory) *ChangeSet {
 
+	if opts.Mode == "" {
+		opts.Mode = ExactChangeSetMode
+	}
```

Comment:
seems like we should be able to make tighter assertion that mode isnt empty instead of just defaulting.

Reply:
It is to preserve original behaviour where empty `ChangeSetOpts{}` used  exact change factory function.  Empty `ChangeSetOpts{}` is used in tests only.